Model/Configuration.php

```diff
@@ -39,81 +39,64 @@ class Configuration
      * @param \Magento\Config\Model\ResourceModel\Config $resourceConfig
      * @param \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
      * @param \Magento\Store\Model\StoreManagerInterface $storeManager,
-     * @param int $storeId The ID of the Store View (called `store` in the DB and code)
+     * @param int $websiteId The ID of the Website
      */
     public function __construct(
         \Magento\Config\Model\ResourceModel\Config $resourceConfig,
         \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
         \Magento\Store\Model\StoreManagerInterface $storeManager,
-        int $storeId
+        int $websiteId
     ) {
         $this->resourceConfig = $resourceConfig;
         $this->scopeConfig = $scopeConfig;
         $this->storeManager = $storeManager;
-        $this->storeId = $storeId;
-
-        // M2 requires a more modern approach of not using 0 as the default ID,
-        // but rather relying on a scope type. This is great, but we don't need
-        // that complexity. Simplify...
-        if ((int) $storeId === \Magento\Store\Model\Store::DEFAULT_STORE_ID) {
-            $this->scope = \Magento\Framework\App\Config\ScopeConfigInterface::SCOPE_TYPE_DEFAULT;
-        } else {
-            $this->scope = \Magento\Store\Model\ScopeInterface::SCOPE_STORE;
-        }
-    }
+        $this->websiteId = $websiteId;
 
-    public function getStoreId()
-    {
-        return $this->storeId;
+		$this->scope = \Magento\Store\Model\ScopeInterface::SCOPE_WEBSITES;
     }
 
-    /**
-     * The website ID attached to the current store view.
-     *
-     * @return int
-     */
     public function getWebsiteId()
     {
-        return $this->storeManager->getStore($this->storeId)->getWebsiteId();
+        return $this->websiteId;
     }
 
     public function getWisUrl()
     {
-        return $this->getStoreConfig(self::WIS_URL_PATH);
+        return $this->getWebsiteConfig(self::WIS_URL_PATH);
     }
 
     public function getIntegrationToken()
     {
-        return $this->getStoreConfig(self::INTEGRATION_TOKEN);
+        return $this->getWebsiteConfig(self::INTEGRATION_TOKEN);
     }
 
     /**
      * @return bool
      */
     public function isEnabled()
     {
-        return (bool) $this->getStoreConfig(self::MODULE_ENABLED_PATH);
+        return (bool) $this->getWebsiteConfig(self::MODULE_ENABLED_PATH);
     }
 
     public function getSalt()
     {
-        return $this->getStoreConfig(self::SALT_PATH);
+        return $this->getWebsiteConfig(self::SALT_PATH);
     }
 
     public function getLogSettings()
     {
-        return $this->getStoreConfig(self::LOG_SETTINGS_PATH);
+        return $this->getWebsiteConfig(self::LOG_SETTINGS_PATH);
     }
 
     /**
      * @param string $path
      */
-    protected function getStoreConfig($path)
+    protected function getWebsiteConfig($path)
```

Comment:
I wonder whether it would make sense to call this `getConfig` since it's kind of redundant…